automated_packaging/prepare_changelog.pl

```diff
@@ -0,0 +1,176 @@
+      . 'https://api.github.com/';
+    my $result    = `$cmd > /dev/null 2>&1`;
+    my $exit_code = $? >> 8;
+
+    if ( $exit_code == 22 ) {
+    	die "Your token was rejected by GitHub.";
+    }
+
+    return $github_token;
+}
+
+# Necessary to write log date
+my @month_names = qw(January February March April May June July August September October November December);
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time);
+$year += 1900;
+
+sub add_changelog_header() {
+	# Update the changelog file
+	open(CHANGELOG_FILE, ">CHANGELOG.md") || die "Can't open the Changelog file !";
+	print CHANGELOG_FILE "### citus v$NEW_VERSION ($month_names[$mon] $mday, $year) ###\n\n";
+}
+
+my $github_token = get_and_verify_token();
+
+# Means it is a major release
+if ($NEW_VERSION =~ /.*\.0$/) {
+	# Checkout the master branch
+	`git checkout master`;
+
+	# Now create a new branch based on master
+	`git checkout -b citus-$NEW_VERSION-changelog`;
+
+	# Read the current changelog
+	open(CHANGELOG_FILE, "<CHANGELOG.md") || die "Changelog file not found";
+	my @changelog_current_lines = <CHANGELOG_FILE>;
+	close(CHANGELOG_FILE);
+
+	add_changelog_header();
+
+	my $page_number=1;
+	do {
+		my $pr_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET 'https://api.github.com/repos/citusdata/citus/pulls?base=master&state=all&page=$page_number'`;
+		my @prs = @{decode_json($pr_text)};
+
+		foreach my $pr ( @prs ) {
+			%pr_hash = %$pr;
+
+			if($pr_hash{'number'} == $NUMBER_OF_LAST_PR) {
+				last;
+			}
+
+			@log_output = split("\n", $pr_hash{'body'});
+			foreach $line (@log_output) {
+				if($line =~ /^DESCRIPTION:.*/) {
+					$description_part=substr $line ,13 ,-1;
+					if(length($description_part) > 78) {
+						`git reset --hard`;
+						die "Can not add description longer than 78 charachters";
+					}
+
+					print CHANGELOG_FILE "* " . $description_part . "\n\n";
+				}
+			}
+		}
+				
+		$page_number += 1;
+
+	} while(1);
+
+	print CHANGELOG_FILE @changelog_current_lines;
+	close(CHANGELOG_FILE);
+
+	# Commit and push the changes, then open a PR
+	`git commit -a -m "Add changelog entry for $NEW_VERSION"`;
+	`git push origin citus-$NEW_VERSION-changelog`;
+	`curl -H "Accept: application/vnd.github.v3.full+json" -X POST --user "$github_token:x-oauth-basic" -d '{"title":"Bump Citus to $NEW_VERSION", "head":"master", "base":"citus-$NEW_VERSION-changelog"}' https://api.github.com/repos/citusdata/citus/pulls`;
+}
+
+# It is a point release
+else {
+	`git checkout master`;
+	`git checkout -b release-$NEW_VERSION-changelog`;
+
+	# Read the current changelog
+	open(CHANGELOG_FILE, "<CHANGELOG.md") || die "Changelog file not found";
+	my @changelog_current_lines = <CHANGELOG_FILE>;
+	close(CHANGELOG_FILE);
+
+	add_changelog_header();
+
+	my $page_number=1;
+	do {
+		my $issues_text = `curl -H "Accept: application/vnd.github.v3.full+json" -X GET 'https://api.github.com/repos/citusdata/citus/issues?state=all&page=$page_number'`;
```

Comment:
This will return all issues regarding they are merged or not. You need to cross check whether it is merged from pulls endpoint [like this](https://developer.github.com/v3/pulls/#get-if-a-pull-request-has-been-merged). It would make sense to put that logic in a sub.